src/sentry/workflow_engine/migration_helpers/issue_alert_migration.py

```diff
@@ -89,14 +90,22 @@ def _create_detector_lookup(self) -> Detector:
                 error_detector = Detector(type=ErrorGroupType.slug, project=self.project)
 
         else:
-            error_detector, _ = Detector.objects.get_or_create(
-                type=ErrorGroupType.slug,
-                project=self.project,
-                defaults={"config": {}, "name": "Error Detector"},
-            )
-            _, created = AlertRuleDetector.objects.get_or_create(
-                detector=error_detector, rule_id=self.rule.id
+            lock = locks.get(
+                f"workflow-engine-project-error-detector:{self.project.id}",
+                duration=10,
+                name="workflow_engine_issue_alert",
             )
+            with lock.acquire():
+                error_detector, _ = Detector.objects.get_or_create(
+                    type=ErrorGroupType.slug,
+                    project=self.project,
+                    defaults={"config": {}, "name": "Error Detector"},
```

Comment:
i think i missed this bit in a previous review -- just curious, should these always be named error detector or is there a name for the issue alert that we could use?

Reply:
the detector is just called "Error Detector" by default